drivers/staging/vc04_services/interface/vchiq_arm/vchiq_core.c

```diff
@@ -587,8 +587,11 @@ reserve_space(VCHIQ_STATE_T *state, int space, int is_blocking)
 				return NULL; /* No space available */
 		}
 
-		BUG_ON(tx_pos ==
-			(state->slot_queue_available * VCHIQ_SLOT_SIZE));
+		if (tx_pos >= (state->slot_queue_available * VCHIQ_SLOT_SIZE)) {
```

Comment:
This is beyond the topic of the patch series. I will keep the logic and only replace BUG_ON with WARN_ON. Would that be okay?